### PR TITLE
feat(testing): add rtl tests for components

### PR DESCRIPTION
**Pull request by yiyang-qantashotels**
- Alert
- ArrowControl
- BackgroundImage
- Badge
- BadgeFrame
- BadgeTest
- BlockLink
- Breadcrumb
- Breadcrumbs
- Button
- Card
- CharacterCount

## Description

💁‍

## Related issues

💪

## Looks like this

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan), [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/Button/Button.test.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import theme from 'theme';
+import { axe, render, screen } from 'testUtils';
+
+import Button from './';
+
+import get from 'lodash/get';
+
+jest.mock('lodash/get');
```
Comment (ajlu) [inline src/components/Button/Button.test.jsx:9]: why are we mocking `get`?

File: src/components/CharacterCount/CharacterCount.test.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { axe, render, screen } from 'testUtils';
+
+import CharacterCount from './';
+
+const props = {
+  value: 'Hello world',
+  limit: 64,
+};
```
Comment (ajlu) [inline src/components/CharacterCount/CharacterCount.test.jsx:9]: can inline this in line 35 given its only used there?

File: src/components/BackgroundImage/BackgroundImage.test.jsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { axe, render, screen } from 'testUtils';
+
+import BackgroundImage from './';
+
+test('renders BackgroundImage element with children', () => {
+  render(
+    <BackgroundImage src="/example.jpg" data-testid="background-image">
```
[the remark this answers is not in the document]
Comment (yiyang-qantashotels) [reply src/components/BackgroundImage/BackgroundImage.test.jsx:8]: changed to role="img" 👍

File: src/components/Badge/Badge.test.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { axe, render, screen } from 'testUtils';
+import Badge from './';
+import {
+  getBadgeFrameProps,
+  getBadgeIconMarginPropByBadgeSize,
+} from './helpers';
+
+jest.mock('./helpers');
```
[the remark this answers is not in the document]
Comment (yiyang-qantashotels) [reply src/components/Badge/Badge.test.jsx:9]: removed all mocks from helpers 👍

File: src/components/CharacterCount/CharacterCount.test.jsx (hunk 2 above)
Comment (yiyang-qantashotels) [reply src/components/CharacterCount/CharacterCount.test.jsx:9]: changed to inline. Thanks🤘

**Review by ajlu — APPROVED** [review]
Great work! 🚀

**Pull request merged**